tests/functional/intelligent-host-selection/05-from-platform-group.t

```diff
@@ -18,16 +18,20 @@
 # Test that Cylc Can select a host from a platform group
 # Failing if there is no good host _any_ platform
 # Succeeding if there is no bad host on any platform in the group
-export REQUIRE_PLATFORM='loc:remote fs:indep comms:tcp'
+export REQUIRE_PLATFORM='loc:remote fs:indep'
 
 . "$(dirname "$0")/test_header"
 
 #-------------------------------------------------------------------------------
 set_test_number 11
 
-create_test_global_config "" "
+# Create `global.cylc`` here rather than use
+# `test_header.create_test_global_config`` method which appends
+# to existing config: Stop users accidentally creating platforms which would
+# match the Cylc 7 settings in `flow.cylc`.
+cat >> 'global.cylc' <<__HERE__
 [platforms]
-    [[${CYLC_TEST_PLATFORM}]]
+    [[mixedhostplatform]]
```

Comment:
If you are not running *any* jobs on the remote platform you can do more or less whatever you like here.

If you *are* running jobs on the remote then this needs to be kept as `$CYLC_TEST_PLATFORM` otherwise you are bypassing the test config.

For example if your `global-tests.cylc` contained this:

```
[platforms]
    [[_remote_background_poll_indep]]
        comms method = poll
        host = abc
```

And your test configured this platform:

```
[platforms]
    [[whatever]]
        hosts = ${CYLC_TEST_HOST}
```

Then any jobs run on "whatever" would *not* use `comms method = poll` which could invalidate the test or cause it to fail for other reasons e.g. `install target` or `cylc executable`.

Reply:
Note that the tests *only* use test platforms (ones starting with `_`). Users should configure these platforms appropriately for testing.